parts/gearbox.go

```diff
@@ -0,0 +1,45 @@
+package parts
+
+import "fmt"
+
+type Gearbox struct {
+	Motor       MotorManipulator
+	Wheels      WheelsManipulator
+	CurrentGear string
+}
+
+func (g *Gearbox) SetMotor(motor MotorManipulator) {
+	g.Motor = motor
+}
+
+func (g *Gearbox) SetWheels(wheels WheelsManipulator) {
+	g.Wheels = wheels
+}
+
+func (g *Gearbox) UpdateWheelsRPM() {
```

Comment:
This is certainly one way to do it.  Looks like you had a plan, and worked in a specific direction here.  That is good, but be careful that all requirements are followed.